### PR TITLE
Docs refresh

### PR DESCRIPTION
**Pull request by sverhoeven**
- Upgrades the doc deps to latest versions.
- nbsphinx + [recommonmark](https://github.com/readthedocs/recommonmark) (is deprecated) -> myst-nb
- no more warnings/errors during build by fixing links and formatting
  - for example you can not link to private or undocumented things

Refs #441 

TODO
* [x] check all links work
* [x] in rst
* [x] in ipynb
  * [x] to api
  * [x] to other ipynb
* [x] in api

### CODE REVIEW

**Comment by sverhoeven** [comment]
Went on a side-quest with [autodoc2](https://github.com/eWaterCycle/ewatercycle/tree/autodoc2-no-google-docstring), but found out it does not support google style dostrings so it was abandoned.

**Comment by sverhoeven** [comment]
Sphinx 8 was release last week, but extenstions have not yet caught up, so sticking to 7.4.7 as determined by conda lock.

File: docs/Makefile [hunk 1]
```diff
@@ -1,10 +1,10 @@
 # Minimal makefile for Sphinx documentation
 #
 
-# You can set these variables from the command line.
-SPHINXOPTS    =
-SPHINXBUILD   = sphinx-build
-SPHINXPROJ    = ewatercycle
```
Comment (Peter9192) [inline docs/Makefile:7]: Is sphinxproj not needed anymore? Can't find docs on it in a quick search

File: src/ewatercycle/analysis/__init__.py [hunk 2]
```diff
@@ -46,38 +48,24 @@ def hydrograph(
...
-        Units for the precipitation data.
-    figsize : (float, float), optional
-        With, height of the plot in inches.
-    filename : str or Path, optional
-        If specified, a copy of the plot will be saved to this path.
-    nbars : Int, optional
-        Number of bars to use for downsampling precipitation.
-    **kwargs:
-        Options to pass to the matplotlib plotting function
+    Args:
```
Comment (Peter9192) [inline src/ewatercycle/analysis/__init__.py:51]: Is it okay to leave this in google style? Or should it be reverted for consistency?

File: src/ewatercycle/esmvaltool/search.py [hunk 3]
```diff
@@ -79,10 +80,10 @@ def search_esgf(
         variables: Which variables are you searching for. Use the short_name definition.
             For example: ['pr', 'tas'].
         project: Which project to search in. Defaults to 'CMIP6'.
-        extended_mip_tables (optional): If you want to use extended MIP tables.
+        extended_mip_tables: Optional. If you want to use extended MIP tables.
             These tables are probably not relevant for most hydrology usecases and can
             make the search slower. Defaults to False.
-        verbose (optional): If the results should be printed in a verbose way, to aid
+        verbose: Ooptional. If the results should be printed in a verbose way, to aid
```
Comment (Peter9192) [inline src/ewatercycle/esmvaltool/search.py]:
```suggestion
        verbose: Optional. If the results should be printed in a verbose way, to aid
```

File: docs/make.bat [hunk 4]
```diff
@@ -9,9 +9,6 @@ if "%SPHINXBUILD%" == "" (
 )
 set SOURCEDIR=.
 set BUILDDIR=_build
-set SPHINXPROJ=ewatercycle
```
Comment (Peter9192) [inline docs/make.bat:12]: same question

**Review by Peter9192 — APPROVED** [review]
PR has many good improvements. Better out-of-the-box handling of notebooks is nice. You've been porting some docstrings from numpy to google, then abandoned autodoc, but not reverted all the docstring style changes. Is that okay?

File: docs/Makefile (hunk 1 above)
Comment (sverhoeven) [reply docs/Makefile:7]: I did

```sh
# ewatercycle env active
# in temp dir
sphinx-quickstart
```

From which I copied the Makefile and make.bat, so indeed the SPHINXPROJ is no longer needed. I think it parses `project` from conf.py

File: docs/make.bat (hunk 4 above)
Comment (sverhoeven) [reply docs/make.bat:12]: see answer Makefile

File: src/ewatercycle/analysis/__init__.py (hunk 2 above)
Comment (sverhoeven) [reply src/ewatercycle/analysis/__init__.py:51]: I checked all docstrings and they are all in google style now.

**Comment by sverhoeven** [comment]
Thanks for reviewing. All docstring are now use google style.

**Pull request merged**